docs/pages/access-controls/guides/device-trust.mdx

```diff
@@ -0,0 +1,326 @@
+cluster:
+
+```yaml
+version: v6
+kind: role
+metadata:
+  name: device-admin
+spec:
+  allow:
+    rules:
+    - resources: ["device"]
+      verbs:
+      - create
+      - read
+      - list
+      - update
+      - delete
+      - create_enroll_token
+      - enroll
+```
+
+```code
+$ tctl create -f device-admin.yaml
+# role 'device-admin' has been created
+```
+
+<Admonition type="tip" title="Editor Role">
+  For clusters created using Teleport v12 or newer, the builtin role `editor`
+  has the necessary permissions to manage devices.
+  For production setups a separate role is still recommended.
+</Admonition>
+
+Note that, in addition to the usual CRUD verbs (create, read, list, update and
+delete), we have also included `create_enroll_token` and `enroll`. The
+`create_enroll_token` verb is necessary to execute the `tctl devices enroll`
+command; `enroll` is necessary to execute `tsh device enroll`. Both are
+demonstrated in the following section.
+
+Make sure your user has the `device-admin` role:
+
+```yaml
+kind: user
+metadata:
+  name: codingllama
+  # unrelated fields omitted.
+spec:
+  roles:
+  - access
+  - device-admin # add this line
+version: v2
+```
+
+```code
+$ tctl edit users/codingllama
+# user "codingllama" has been updated
+$ tsh logout; tsh login --proxy=example.com --user=codingllama ## fetch new certificates
+```
+
+Having the necessary permissions, let's add a device to the inventory. Replace
+the `$serial` below with your macOS device serial number (visible under Apple ->
+"About This Mac" -> "Serial number").
+
+```code
+$ serial="$(ioreg -c IOPlatformExpertDevice -d 2 | grep -i IOPlatformSerialNumber | awk -F'"' '{print $4}')"
+$ tctl devices add --os=macos --asset-tag="$serial"
+# Device C00AA3BBBB6C/macOS added to the inventory
+```
+
+View registered devices:
+
+```code
+$ tctl devices ls
+# Asset Tag    OS    Enroll Status Device ID
+# ------------ ----- ------------- ------------------------------------
+# C00AA3BBBB6C macOS not enrolled  d40f2ee4-856d-4aef-b784-c4371e39c036
+```
+
+A registered device is a device that Teleport knows about. Once a registered
+device is enrolled, it becomes a trusted device.
+
+<Admonition type="tip" title="Removing Devices">
+  A device that is no longer in use may be removed using
+  `tctl devices rm --device-id=<ID>` or
+  `tctl devices rm --asset-tag=<SERIAL_NUMBER>`.
+</Admonition>
+
+## Step 2/2. Enroll a trusted device
+
+A registered device becomes a trusted device after it goes through the
+enrollment ceremony. Enrollment exchanges an enrollment token, created by a
+device admin, for the opportunity to enroll the corresponding device.
+
+To create an enrollment token run the following command, where `--asset-tag` is
+the serial number of the device we want to enroll.
+
+```code
+$ tctl devices enroll --asset-tag="C00AA3BBBB6C"
+# Run the command below on device "C00AA3BBBB6C" to enroll it:
+# tsh device enroll --token=MQi9gkOfyvdyNyN2moJZz9cqx/GyILR5uNu4NE9dxO8
```

Comment:
```suggestion
Run the command below on device "C00AA3BBBB6C" to enroll it:
tsh device enroll --token=MQi9gkOfyvdyNyN2moJZz9cqx/GyILR5uNu4NE9dxO8
```

Reply:
We should consider if we want to create a preset enrollment token in `config.json` that's obviously fake and use it here and anywhere else we end up talking about device enrollment.

Reply:
Enrollment tokens are random blobs, there's nothing special encoded on them. Happy to use a common token if you think it's best, though.

Reply:
Only because we've been told int the past (can't find the issue/PR at the moment) that we get false-positive security reports for having example tokens that look real enough.

Reply:
Done for asset tag and enroll tokens, all files/occurrences.